### PR TITLE
DCS-402, DCS-398, DCS-401

### PR DESCRIPTION
**Pull request by cergina**
Combined three tasks because of their shortness

- Add ability to view flat via tabs
- Fix bug that causes empty central unit visit look horrible
- Fix bug making it possible to create Central Unit on same dip address

### CODE REVIEW

File: src/main/java/Control/Scenario/UC_NewCentralUnit.java [hunk 1]
```diff
@@ -120,11 +124,18 @@ private void insertIntoDatabase(@NotNull final CentralUnit centralUnit) throws S
...
+        try {
+            return I_CentralUnit.checkIfExists(db.getConn(), db.getPs(), db.getRs(), centralUnit.getUid(), centralUnit.getDipAddress(), centralUnit.getBuildingId());
+        } catch (SQLException sqle) {
+            CustomLogs.Error(sqle.getMessage());
+        }
+
+        return true;
+    }
 
+    private boolean checkIfDipNotFree(@NotNull final String dip) {
```
Comment (ludko) [inline src/main/java/Control/Scenario/UC_NewCentralUnit.java:136]: akoze ja mam rad ked je to na styl isDipFree povedzme nie negacie spojky a pod.

File: src/main/webapp/WEB-INF/templates/views/privileged/my_flat.html [hunk 2]
```diff
@@ -30,93 +39,102 @@ <h4>LAST 30 DAYS</h4>
...
-                    <!-- HIDE THIS BUT KEEP THIS ITS IMPORTANT-->
-                    <input type="hidden" th:value="${flat.id}" id="controller-flat-id">
-                    <!-- BUTTONS FOR ADDITION -->
-                    <div class="row justify-content-center rowinrightcol">
-                        <div class="regpaddingbottom padding-top-10">
-                            <button type="submit" class="roundedbutton min-width-300" onclick="createControllerUnitForThisFlat()">Create</button>
+
+                        <!-- list opf controller units -->
+                        <div th:unless="${flat.controllerUnits.isEmpty()}">
+                            <!-- test only -->
```
Comment (ludko) [inline src/main/webapp/WEB-INF/templates/views/privileged/my_flat.html:124]: akoze tie test only komentare by si mohol odstranit z celeho projektu

File: src/main/java/Control/Scenario/UC_NewCentralUnit.java [hunk 3]
```diff
@@ -38,6 +38,10 @@ public UC_NewCentralUnit(@NotNull DbProvider dbProvider) {
                 throw new AlreadyExistsException("Central Unit already exists.");
             }
 
+            if(checkIfDipNotFree(centralUnit.getDipAddress())){
+                throw new AlreadyExistsException("Dip address of central unit already occupied. Use different");
```
Comment (xcipelova) [inline src/main/java/Control/Scenario/UC_NewCentralUnit.java]: tu by som doplnila Use different *DIP address*

File: src/main/java/Control/Scenario/UC_NewCentralUnit.java (hunk 1 above)
Comment (xcipelova) [reply src/main/java/Control/Scenario/UC_NewCentralUnit.java:136]: pripajam sa, je to kratsie

File: src/main/java/Model/Database/Interaction/I_CentralUnit.java [hunk 4]
```diff
@@ -153,7 +151,7 @@ public static boolean checkIfExists(Connection conn, PreparedStatement ps, Resul
                 "SELECT " +
                         "* " +
                         "FROM " + T_CentralUnit.DBTABLE_NAME + " " +
-                        "WHERE Uid=? OR (BuildingID=? AND DipAddress=?)"
+                        "WHERE Uid=? OR (BuildingID=? AND DipAddress=?) "
```
Comment (xcipelova) [inline src/main/java/Model/Database/Interaction/I_CentralUnit.java:154]: naco je tu medzera na konci?

File: src/main/webapp/WEB-INF/templates/views/privileged/my_flat.html [hunk 5]
```diff
@@ -30,93 +39,102 @@ <h4>LAST 30 DAYS</h4>
...
                         </div>
-                    </div>
-                    <!-- HIDE THIS BUT KEEP THIS ITS IMPORTANT-->
-                    <input type="hidden" th:value="${flat.id}" id="controller-flat-id">
-                    <!-- BUTTONS FOR ADDITION -->
-                    <div class="row justify-content-center rowinrightcol">
-                        <div class="regpaddingbottom padding-top-10">
-                            <button type="submit" class="roundedbutton min-width-300" onclick="createControllerUnitForThisFlat()">Create</button>
+
+                        <!-- list opf controller units -->
```
Comment (xcipelova) [inline src/main/webapp/WEB-INF/templates/views/privileged/my_flat.html]: tu mas asi typo, list *of*

**Review by xcipelova — COMMENTED** [review]
Kod je v poriadku

**Pull request merged**